src/frontend-build.sh

```diff
@@ -67,6 +76,33 @@ else
   echo "Publishing to Quay without expiration"
 fi
 
-curl -sSL $COMMON_BUILDER/src/quay_push.sh | bash -s
+# source <(curl -sSL $COMMON_BUILDER/src/quay_push.sh | bash -s)
+
+# IMAGE, IMAGE_TAG, and Tokens are exported from upstream pr_check.sh
+export LC_ALL=en_US.utf-8
+export LANG=en_US.utf-8
+export APP_ROOT=${APP_ROOT:-pwd}
+export WORKSPACE=${WORKSPACE:-$APP_ROOT}  # if running in jenkins, use the build's workspace
+# export IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+export GIT_COMMIT=$(git rev-parse HEAD)
+
+
+if [[ -z "$QUAY_USER" || -z "$QUAY_TOKEN" ]]; then
+    echo "QUAY_USER and QUAY_TOKEN must be set"
+    exit 1
+fi
+
+if [[ -z "$RH_REGISTRY_USER" || -z "$RH_REGISTRY_TOKEN" ]]; then
+    echo "RH_REGISTRY_USER and RH_REGISTRY_TOKEN must be set"
+    exit 1
+fi
+
+
+DOCKER_CONF="$PWD/.docker"
+mkdir -p "$DOCKER_CONF"
+docker --config="$DOCKER_CONF" login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
+docker --config="$DOCKER_CONF" login -u="$RH_REGISTRY_USER" -p="$RH_REGISTRY_TOKEN" registry.redhat.io
```

Comment:
It's unsafe to use `-p` with docker, it's much better to use `--password-stdin` https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin ideally we'd store these passwords in file and read from them. But we can run `echo $QUAY_TOKEN | docker --config="$DOCKER_CONF" login -u="$QUAY_USER" --password-stdin quay.io`

Reply:
i just copy-pasted from the existing quay_push.sh, but i'm open to change it